meetings/2021-08/sept-01.md

```diff
@@ -0,0 +1,585 @@
+
+ATA: Yeah, it is concerning. So in two slides, there was a slide about what’s in scope in terms of the direction of the interop here. So I think at this point with very minimal proposal, the intention is that the interop would mainly be about explaining how these WasmGC structs get reflected as as JS object, and I think where a lot of the type system issues come in is where we want to do the import from a JS [unable to transcribe]. And in particular, I think there’s also a difference between, you know, when we talk about round tripping. So if you have a Wasm struct that goes to JS and back, that’s probably possible, but if you were to create something entirely in JS, then import it into Wasm, then you have to have some type information for this to be possible. And so I think that’s where these future extensions would fit in? And that’s I think where this issue of having types, which come in if I understand your concern. Does that answer that?
+
+YSV: Yeah, that does more or less answer it. Now I can’t speak for our Wasm team, they have very limited time in terms of giving this proposal the amount of time that it needs. So they haven’t been able to come back to me with regards to any further specific issues they have here. But what I would like to say is I’m perfectly fine with going to Stage 1. I want to spend some time working with a few people not only on the direct Wasm team, but also adjacent to it. So ATA would be one person, but I also want to speak with Luke Wagner and a couple of other folks to get a bit more feedback here before we would look at something like Stage 2. The WebAssembly team on Mozilla’s side, on SpiderMonkey, is a little uncomfortable with how quickly this is moving forward. And I will try to get some information soon, but I can’t promise how soon that will be. So if you’re looking to move this quickly to Stage 2, I would ask that we work closely together on the pacing of it.
+
+SYG: I hear you and I intend to work closely with you. To the urgency question and the speed that I’m envisioning. This slide [slide 22, about strings] is I think the actual thing that might block implementations for a significant amount of time, and that is what I’m working on, and that is not blocked by, you know, standards progress. I think I want to get the ball rolling here. There are many interested parties and let’s try to nail down some design. That is amenable to everybody while this part which I think is the hard part is happening. 
+
+YSV: Yeah, for sure. For this, I’m going to have our GC folks take a look and work with you on that once they’ve got some cycles to do that.
+
+JWK: Currently on the web, the concurrent programming model is based on posting messages instead of memory sharing. So, adding a high-level abstraction of shared structs…that means we are encouraging the concurrent programming model based on memory sharing. 
```

Comment:
```suggestion
JWK: Currently on the web, the concurrent programming model is based on post messages instead of memory sharing. Adding a high-level abstraction of shared structs… That means we are encouraging the concurrent programming model based on memory sharing. 
```